### PR TITLE
[FEATURE] #99834 - Add two PSR-14 events around auto-create redirects

### PR DESCRIPTION
**Pull request by brotkrueml**
Resolves: https://github.com/TYPO3-Documentation/Changelog-To-Doc/issues/383
Releases: main

### CODE REVIEW

File: Documentation/ApiOverview/Events/Events/Redirects/AfterAutoCreateRedirectHasBeenPersistedEvent.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+============================================
+AfterAutoCreateRedirectHasBeenPersistedEvent
+============================================
+
+..  versionadded:: 12.3
+
+The PSR-14 event
+:php:`\TYPO3\CMS\Redirects\Event\AfterAutoCreateRedirectHasBeenPersistedEvent`
+allows extension authors to react on persisted auto-created redirects. This
```
Comment (linawolf) [inline Documentation/ApiOverview/Events/Events/Redirects/AfterAutoCreateRedirectHasBeenPersistedEvent.rst:14]:
```suggestion
allows extensions to react on persisted auto-created redirects. This
```

Comment (linawolf) [inline Documentation/ApiOverview/Events/Events/Redirects/AfterAutoCreateRedirectHasBeenPersistedEvent.rst:14]:
```suggestion
allows extension authors to react on persisted auto-created redirects. This event
```

File: Documentation/ApiOverview/Events/Events/Redirects/AfterAutoCreateRedirectHasBeenPersistedEvent.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+can be used to call external APIs or perform other tasks based on the real
+persisted redirects.
+
+..  note::
+    To handle later updates or react on manual created redirects in the backend
```
Comment (linawolf) [inline Documentation/ApiOverview/Events/Events/Redirects/AfterAutoCreateRedirectHasBeenPersistedEvent.rst:19]:
```suggestion
    To handle later updates or react on manually created redirects in the backend
```

File: Documentation/ApiOverview/Events/Events/Redirects/ModifyAutoCreateRedirectRecordBeforePersistingEvent.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+===================================================
+ModifyAutoCreateRedirectRecordBeforePersistingEvent
+===================================================
+
+..  versionadded:: 12.3
+
+The PSR-14 event
+:php:`\TYPO3\CMS\Redirects\Event\ModifyAutoCreateRedirectRecordBeforePersistingEvent`
+allows extension authors to modify the redirect record before it is persisted to
```
Comment (linawolf) [inline Documentation/ApiOverview/Events/Events/Redirects/ModifyAutoCreateRedirectRecordBeforePersistingEvent.rst:14]:
```suggestion
allows extensions to modify the redirect record before it is persisted to
```

File: Documentation/ApiOverview/Events/Events/Redirects/ModifyAutoCreateRedirectRecordBeforePersistingEvent.rst [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+the database. This can be used to change values according to circumstances, such
+as, for example, different sub-tree settings that are not covered by the Core
```
Comment (linawolf) [inline Documentation/ApiOverview/Events/Events/Redirects/ModifyAutoCreateRedirectRecordBeforePersistingEvent.rst:16]:
```suggestion
as different sub-tree settings that are not covered by the Core
```

File: Documentation/ApiOverview/Events/Events/Redirects/ModifyAutoCreateRedirectRecordBeforePersistingEvent.rst [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+site configuration. Another use case could be to write data to additional
+:sql:`sys_redirect` columns added by a custom extension for later use.
+
+..  note::
+    To handle later updates or react on manual created redirects in the backend
```
Comment (linawolf) [inline Documentation/ApiOverview/Events/Events/Redirects/ModifyAutoCreateRedirectRecordBeforePersistingEvent.rst:21]:
```suggestion
    To handle updates or react on manual created redirects in the backend
```

File: Documentation/ApiOverview/Events/Events/Redirects/AfterAutoCreateRedirectHasBeenPersistedEvent.rst (hunk 1 above)
Comment (brotkrueml) [reply Documentation/ApiOverview/Events/Events/Redirects/AfterAutoCreateRedirectHasBeenPersistedEvent.rst:14]: How can a thing react?

Comment (linawolf) [reply Documentation/ApiOverview/Events/Events/Redirects/AfterAutoCreateRedirectHasBeenPersistedEvent.rst:14]: How can the extension author react when he is not there?

Comment (brotkrueml) [reply Documentation/ApiOverview/Events/Events/Redirects/AfterAutoCreateRedirectHasBeenPersistedEvent.rst:14]: The developer reacts on an event programmatically...

Comment (linawolf) [reply Documentation/ApiOverview/Events/Events/Redirects/AfterAutoCreateRedirectHasBeenPersistedEvent.rst:14]: I would have said, the developer instructs the extension to react. But whatever. Keep it like this if you want

**Review by linawolf — APPROVED** [review]

**Pull request merged**